### PR TITLE
parametric: Add tests for OTEL

### PR DESCRIPTION
**Pull request by dianashevchenko**
## Description

Adds a bunch of tests for OTEL APIs in the tracers. To start only the `go` library runs these tests. In reviewing, pay the most attention to ensuring the expectations are correct as this is the first set of tests here.

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

**Comment by cbeauchesne** [comment]
-> https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md ;)

**Comment by cbeauchesne** [comment]
Do you want it to merge in `shared/otel-parametric` or in `main` ?

File: parametric/apps/golang/main.go
Comment (ajgajg1134) [inline parametric/apps/golang/main.go:32]: Doesn't this also happen in `StartTracer`? (Is it intentional to have the otel trace provider set in two different places for different kinds of tests?)

File: parametric/test_otel_span_methods.py
Comment (ajgajg1134) [inline parametric/test_otel_span_methods.py:203]: did you mean to skip / ignore this test?

File: parametric/apps/golang/otel_methods.go
Comment (katiehockman) [inline parametric/apps/golang/otel_methods.go:252]: Should probably panic if an error occurs here, instead of ignoring it.

File: parametric/test_otel_span_methods.py
Comment (katiehockman) [inline parametric/test_otel_span_methods.py:36]: can we set the name and the resource different things, to make sure they're being set correctly, or is this just how the OTel API works?

File: parametric/test_otel_span_methods.py
Comment (katiehockman) [inline parametric/test_otel_span_methods.py:48]:
```suggestion
```
Looks like it's duplicated?

File: parametric/test_otel_span_methods.py
Comment (katiehockman) [inline parametric/test_otel_span_methods.py:80]:
```suggestion
def test_otel_span_is_recording(test_agent, test_library):
```

File: parametric/test_otel_tracer.py
Comment (katiehockman) [inline parametric/test_otel_tracer.py:63]: Should we look into this TODO in a follow-up PR then?

**Review by katiehockman — APPROVED** [review]
I didn't check every test super carefully yet, but overall it looks great. Thanks!

File: parametric/apps/golang/main.go
Comment (dianashevchenko) [reply parametric/apps/golang/main.go:32]: I have removed StartTracer in this iteration

File: parametric/apps/golang/main.go
Comment (dianashevchenko) [reply parametric/apps/golang/main.go:127]: In newServer()

File: parametric/test_otel_span_methods.py
Comment (dianashevchenko) [reply parametric/test_otel_span_methods.py:36]: This is technically how _Datadog_ works, if there's no resource specified with Datadog StartSpanOptions. So, simple OTel span would have same name and resource

File: parametric/test_otel_tracer.py
Comment (dianashevchenko) [reply parametric/test_otel_tracer.py:63]: That's Andrews comment, removed

**Review by pawelchcki — APPROVED** [review]
LGTM but - it would be good if either @Kyle-Verhoog or @cbeauchesne also would take a peek before this is merged

**Review by Qard — COMMENTED** [review]
Overall LGTM. I'll try it out with Node.js soon. 🙂

**Comment by cbeauchesne** [comment]
I can approve it, but @@Kyle-Verhoog  will have a way better feedback on parametric test than mine

**Review by cbeauchesne — APPROVED** [review]

File: parametric/_library_client.py
Comment (Kyle-Verhoog) [reply parametric/_library_client.py:198]: bump here

File: parametric/test_otel_span_with_w3c.py
Comment (Kyle-Verhoog) [inline parametric/test_otel_span_with_w3c.py:54]: stale comment? can this be removed?

File: parametric/test_otel_tracer.py
Comment (Kyle-Verhoog) [inline parametric/test_otel_tracer.py:8]: not sure what this comment means, is there an action to take here still?

**Review by Kyle-Verhoog — COMMENTED** [review]
just a few nits but otherwise looks good to me.

@pawelchcki we should really look into changing over to http sooner rather than later since we're accumulating some debt here.

@dianashevchenko if you're up for it, we should also be implementing otel for the [http interface](https://github.com/DataDog/system-tests/tree/main/parametric#http) as we're looking to get rid of grpc in order to support all the languages.

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Pull request merged**
